mmv1/products/networksecurity/InterceptDeployment.yaml

```diff
@@ -0,0 +1,121 @@
+# Copyright 2024 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+name: 'InterceptDeployment'
```

Comment:
I'd suggest breaking these into separate PRs per-resource, google-private has a lower bar for review at the moment and means re-review is required on the private -> beta promotion. Grouping by pairs (like InterceptDeployment and InterceptDeploymentGroup) _may_ be fine if both are small.

Reply:
The issue with this is that these resources are not independent, for example you can't create an endpoint group association without an endpoint group, and you can't create an endpoint group without a deployment group. Same for deployments. So in order for tests to work the changes would have to go together or stacked.

We can split Mirroring* and Intercept* resource since those are independent, the reason we did combine them in one PR is that their implementation is basically identical, so it made sense to reduce the toll and not have two reviewers reviewing basically the same thing (i.e., every comment will apply to both).

WDYT?